ocs_ci/deployment/rosa.py

```diff
@@ -75,13 +75,11 @@ def destroy(self, log_level="DEBUG"):
             log_level (str): log level openshift-installer (default: DEBUG)
 
         """
+        cluster_details = ocm.get_cluster_details(self.cluster_name)
+        cluster_id = cluster_details.get("id")
         ocm.destroy_cluster(self.cluster_name)
```

Comment:
In destroy function needs to be timeoutsampler that waits until the cluster is really deleted from ocm and after that the leftover functions could be executed.